code/modules/projectiles/guns/energy/kinetic_accelerator.dm

```diff
@@ -323,14 +323,15 @@
 	else if(istype(KA.loc, /mob/living/basic/mining_drone))
 		to_chat(user, span_notice("The modkit you're trying to install is not rated for minebot use."))
 		return FALSE
-	if(denied_type)
-		var/number_of_denied = 0
-		for(var/obj/item/borg/upgrade/modkit/modkit_upgrade as anything in KA.modkits)
-			if(istype(modkit_upgrade, denied_type))
-				number_of_denied++
-			if(number_of_denied >= maximum_of_type)
-				. = FALSE
-				break
+	if(!denied_type)
+		denied_type = src
```

Comment:
Create a new variable within the proc, don't change the variable on the object. This will be retained after this proc runs which is not what we want.

Reply:
Something like
```var/current_denied_types = denied_types```
which we define before these if statements and then do all of our checks using this var instead.